app/controllers/foreman_deployments/api/v2/stacks_controller.rb

```diff
@@ -20,6 +20,18 @@ def create
           process_response @stack.save
         end
 
+        api :PUT, '/stacks/:id/', N_('Update imported stack')
+        param_group :stack, :as => :update
+        param :id, :identifier, :required => true
+        def update
+          if !@stack.configurations.empty? && !params[:stack][:definition].nil?
+            render :json => { :error => _("Can't update stack that has been configured") }, :status => :unprocessable_entity
+          else
+            ForemanDeployments::StackParser.parse(params[:stack][:definition]) if params[:stack][:definition]
```

Comment:
Same as above, the reason of this line is input validation. Let's move it to `Stack` model.